dbt_platform_helper/providers/cache.py

```diff
@@ -1,17 +1,54 @@
 import os
+from abc import ABC
+from abc import abstractmethod
 from datetime import datetime
 
+from dbt_platform_helper.providers.aws.interfaces import AwsGetVersionProtocol
 from dbt_platform_helper.providers.yaml_file import YamlFileProvider
 
 
+class CacheStrategy(ABC):
```

Comment:
Are we using an abstract class definition over a protocol here for contract enforcement?

Reply:
yea, I think this is best for Strategies as they will be used for one provider and should implement the same function and no more. Happy to hear thoughts